File: source-connector/src/main/java/sourceconnector/domain/pipeline/Pipeline.java
package sourceconnector.domain.pipeline;

/**
 * Pipeline consists of multiple processors and can get result
 * @param <T> Handling type
 */
public interface Pipeline<T> {
  T getResult();

  /**
   * Move the pipeline to the given offset position
   * @param offset
   * @throws IllegalArgumentException when the given offset is negative
   */
  void toPosition(long offset);
  boolean isComplete();
}

File: source-connector/src/main/kotlin/sourceconnector/domain/pipeline/Pipeline.kt
package sourceconnector.domain.pipeline

/**
 * Pipeline consists of multiple processors and can get result
 * @param <T> Handling type
</T> */
interface Pipeline<T> : AutoCloseable {
  var isComplete: Boolean

  fun getResult(): T?
  /**
   * Move the pipeline to the given offset position
   * @param offset
   * @throws IllegalArgumentException when the given offset is negative
   */
  fun toPosition(offset: Long)
}
